test/ValidationFramework/Validation.hpp

```diff
@@ -0,0 +1,125 @@
+/// Apache License 2.0
+
+#include <iostream>
+
+#include <Global.test.hpp>
+
+class Validation : public ::testing::Test
+{
+   public:
+    enum class OutputQuantity
+    {
+        CARTESIAN_POSITION,
+        CARTESIAN_VELOCITY,
+        MASS,
+        ATMOSPHERIC_DENSITY,
+        THRUST_ACCELERATION,
+    };
+
+    enum class Tool
+    {
+        GMAT,
+        OREKIT,
+    };
+
+    struct Comparison
+    {
+        Tool tool_;
+        std::map<OutputQuantity, double> toleranceMap_;
+    };
```

Comment:
In the future, we can imagine defining different types of comparisons: one that is active for all states saved, one that needs to be checked only at the end of the propagation etc

Reply:
It depends, in trajectory optimization or solvers, the intermediate steps might be different for instance and diverge a bit for while depending on the heuristics, anyway that's for later, no need to change now

Reply:
Yeah, but I think the scope of this specific app should stay as a cross validation platform, and we can see what our needs in terms of self-validation are in the future for the solvers.